internal/installation/paths.go

```diff
@@ -63,11 +67,18 @@ func BinPathFromInstallPath(installPath string) (string, error) {
 }
 
 func InstalledOnPath(installPath string) (bool, string, error) {
```

Comment:
This seems to be containing legacy logic again. As per our new logic the path would ALWAYS be `installPath/bin`, so we don't need to be looking at the StateApp exec path.

If we rely on this function for detecting legacy executables then we should split that off into separate logic that is ONLY used for the update logic. We don't want to sort of "infect" our main application logic with this legacy handling.